pkg/catalog/service.go

```diff
@@ -143,6 +144,7 @@ func (mc *MeshCatalog) GetTargetPortToProtocolMappingForService(svc service.Mesh
 // GetPortToProtocolMappingForService returns a mapping of the service's ports to their corresponding application protocol,
 // where the ports returned are the ones used by downstream clients in their requests. This can be different from the ports
 // actually exposed by the application binary, ie. 'spec.ports[].port' instead of 'spec.ports[].targetPort' for a Kubernetes service.
+// TODO(whitneygriffith): rm or reference provider
 func (mc *MeshCatalog) GetPortToProtocolMappingForService(svc service.MeshService) (map[uint32]string, error) {
```

Comment:
```func (mc *MeshCatalog) GetPortToProtocolMappingForService(svc service.MeshService) (map[uint32]string, error) {
	portToProtocolMap := make(map[uint32]string)

	for _, provider := range mc.serviceProviders {
		currentPortToProtocolMap, err := provider.GetTargetPortToProtocolMappingForService(svc)
		if err != nil {
			return nil, err
		}
		for key, value := range currentPortToProtocolMap {
			portToProtocolMap[key] = value
		}
	}

	if portToProtocolMap == nil {
		return nil, errors.Errorf("Error fetching port:protocol mapping for service %s", svc)
	}

	return portToProtocolMap, nil
}```

Reply:
Will a svc exist in more than one provider, and have differing protocols? 

If the svc exists in more than one providers, and they both have the same ports:
- If port is the same and  protocol is the same, no problem
- If port is the same and protocol, how to handle?